apps/myaccount/src/components/shared/mobile-update-wizard.tsx

```diff
@@ -0,0 +1,421 @@
+                                    "userPortal:components.profile.forms." +
+                                    "generic.inputs.validations.invalidFormat",
+                                    {
+                                        fieldName
+                                    }
+                                ));
+                                validation.isValid = false;
+                            }
+                        } }
+                        value={ currentMobileNumber }
+                    />
+                </Forms>
+            </>
+        );
+    };
+
+    /**
+     * This renders the code verification content.
+     */
+    const renderVerifyCode = (): JSX.Element => {
+        return (
+            <>
+                {
+                    verificationError
+                        ? (
+                            <>
+                                <Message error>
+                                    { t("userPortal:components.mobileUpdateWizard.verifySmsOtp.error") }
+                                </Message>
+                            </>
+                        )
+                        : null
+                }
+                {
+                    resendSuccess
+                        ? (
+                            <>
+                                <Message success>
+                                    { t("userPortal:components.mobileUpdateWizard.notifications." +
+                                        "resendSuccess.message") }
+                                </Message>
+                            </>
+                        )
+                        : null
+                }
+                <Forms
+                    onSubmit={ (values: Map<string, string>) => {
+                        verifyCode(values.get("code"));
+                    } }
+                    submitState={ submit }
+                >
+                    <Field
+                        name="code"
+                        label={ t("userPortal:components.mobileUpdateWizard.verifySmsOtp.label") }
+                        placeholder={ t("userPortal:components.mobileUpdateWizard.verifySmsOtp.placeholder") }
+                        type="text"
+                        required={ true }
+                        requiredErrorMessage={ t("userPortal:components.mobileUpdateWizard.verifySmsOtp." +
+                            "requiredError") }
+                    />
+                    <Segment textAlign="center" basic>
+                        <p className="link" onClick={ resendOTOCode }>{ t("userPortal:components." +
+                            "mobileUpdateWizard.verifySmsOtp.generate") }</p>
+                    </Segment>
+                </Forms>
+            </>
+        );
+    };
+
+    /**
+     * This renders the success message at the end of the OTP flow.
+     */
+    const renderSuccess = (): JSX.Element => {
+        return (
+            <Segment basic textAlign="center">
```

Comment:
We already have an reusable icon for. Shall we use it here?

Reply:
ATM we do not have a reusable icon for this.

Reply:
Yes, let's do this as a later improvement.